# IPUMS_LA_98-SMALL

In [1]:
import os
import pandas as pd

In [2]:
%pwd

'C:\\Users\\Administrator'

In [3]:
%cd C:\Users\Administrator\Documents\datasets

C:\Users\Administrator\Documents\datasets


In [4]:
df=pd.read_csv('ML dataset.csv',sep=",")# IMPORTING THE DATA
df.head()# CHECKING THE FIRST TOP DATA)

year                                                gq         gqtypeg  \
0    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
1    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
2    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
3    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
4    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   

       farm                      ownershg   value  rent  ftotinc  nfams  \
0  Non-Farm                        Rented  999999   325     9505      1   
1  Non-Farm                        Rented  999999   287     8005      1   
2  Non-Farm  Owned_or_being_bought_(loan)   85000     0    29635      1   
3  Non-Farm  Owned_or_being_bought_(loan)  999999     0    52895      1   
4  Non-Farm                        Rented  999999   185     7610      2   

   ncouples  ...  incfarm  incss  incwelfr  incother  poverty  \
0         1  ...        0      0         0         0      128   
1         0  ...        0      0         0         0      211   
2         1  ...   999999  99999     99999     99999      304   
3         1  ...        0      0         0         0      501   
4         0  ...        0      0         0         0      119   

                           migrat5g    migplac5  movedin     vetstat  \
0  Same_state_countydifferent_house  California        0  No_Service   
1  Same_state_countydifferent_house  California        1  No_Service   
2                        Same_house  Same_house        0  No_Service   
3                        Same_house  Same_house        0  No_Service   
4  Same_state_countydifferent_house  California        0  No_Service   

           tranwork  
0  Bus_or_streetcar  
1       Walked_only  
2              Auto  
3       Walked_only  
4              Auto  

[5 rows x 61 columns]

## EXPLOATORY DATA ANALYSIS

In [5]:
#CHECKING THE NO OF ROWS AND COLUMNS
df.shape 

(7485, 61)

In [6]:
#CHECKING THE DATATYPES 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7485 entries, 0 to 7484
Data columns (total 61 columns):
year        7485 non-null int64
gq          7485 non-null object
gqtypeg     7485 non-null object
farm        7485 non-null object
ownershg    7485 non-null object
value       7485 non-null int64
rent        7485 non-null int64
ftotinc     7485 non-null int64
nfams       7485 non-null int64
ncouples    7485 non-null int64
nmothers    7485 non-null int64
nfathers    7485 non-null int64
momloc      7485 non-null int64
stepmom     7485 non-null int64
momrule     7485 non-null int64
poploc      7485 non-null int64
steppop     7485 non-null int64
poprule     7485 non-null int64
sploc       7485 non-null int64
sprule      7485 non-null int64
famsize     7485 non-null int64
nchild      7485 non-null int64
nchlt5      7485 non-null int64
famunit     7485 non-null int64
eldch       7485 non-null int64
yngch       7485 non-null int64
nsibs       7485 non-null int64
relateg     7485 non-null

In [7]:
#CHECKING THE MISSING VALUES
df.isnull().sum()

year        0
gq          0
gqtypeg     0
farm        0
ownershg    0
value       0
rent        0
ftotinc     0
nfams       0
ncouples    0
nmothers    0
nfathers    0
momloc      0
stepmom     0
momrule     0
poploc      0
steppop     0
poprule     0
sploc       0
sprule      0
famsize     0
nchild      0
nchlt5      0
famunit     0
eldch       0
yngch       0
nsibs       0
relateg     0
age         0
sex         0
           ..
marst       0
chborn      0
bplg        0
school      0
educrec     0
schltype    0
empstatg    0
labforce    0
occ1950     0
occscore    0
sei         0
ind1950     0
classwkg    0
wkswork2    0
hrswork2    0
yrlastwk    0
workedyr    0
inctot      0
incwage     0
incbus      0
incfarm     0
incss       0
incwelfr    0
incother    0
poverty     0
migrat5g    0
migplac5    0
movedin     0
vetstat     0
tranwork    0
Length: 61, dtype: int64

#### CHECKING NO OF CATEGORIES PRESENT IN EACH COLUMN

In [8]:
df['year'].value_counts()

98    7485
Name: year, dtype: int64

In [9]:
df['gq'].value_counts()

HH_in_1970_definition_but_sampled_as_larger_unit    7353
Institution                                           73
Other_group_quarters                                  59
Name: gq, dtype: int64

In [10]:
df['gqtypeg'].value_counts()

NA_(household)                     7353
Other_institutions                   55
College_dormitory                    29
Other_non-instit_GQ_and_unknown      21
Correctional_institutions            11
Mental_institutions                   7
Rooming_house                         6
Military                              3
Name: gqtypeg, dtype: int64

In [11]:
df['farm'].value_counts()

Non-Farm    7482
Farm           3
Name: farm, dtype: int64

In [12]:
df['ownershg'].value_counts()

Owned_or_being_bought_(loan)    4169
Rented                          3316
Name: ownershg, dtype: int64

In [13]:
df['value'].value_counts()

999999    3942
112500     450
85000      366
200000     349
95000      334
137500     326
175000     306
77500      243
67500      215
72500      183
62500      173
52500      140
57500       94
42499       78
47500       68
37500       55
32500       43
23750       27
26250       22
21250       18
18750       16
28750       13
16250       10
12500       10
5000         4
Name: value, dtype: int64

In [14]:
df['rent'].value_counts()

0      4305
325     358
212     332
262     283
237     278
375     262
450     252
287     245
500     173
155     135
175     129
185     113
165     112
195      97
135      71
125      66
145      63
105      54
1        41
115      30
75       24
85       21
95       16
65       14
25        6
55        5
Name: rent, dtype: int64

In [15]:
df['ftotinc'].value_counts()

75000     174
999999    132
0         109
12005      63
15005      62
8005       49
4005       45
20005      45
5005       43
10005      42
6005       40
13005      36
18005      35
7005       32
16005      29
17005      28
3005       27
9005       27
24010      26
14005      25
20010      25
11005      25
14010      24
26010      23
18010      22
25010      22
27010      21
15010      21
25005      20
21010      20
         ... 
74410       1
6825        1
8870        1
23205       1
10915       1
25250       1
41630       1
2715        1
19155       1
4820        1
15065       1
41690       1
19215       1
33550       1
29440       1
2815        1
17150       1
45820       1
6905        1
35575       1
8950        1
61330       1
37620       1
10995       1
2795        1
31465       1
21220       1
8930        1
27355       1
65510       1
Name: ftotinc, Length: 3594, dtype: int64

In [16]:
df['nfams'].value_counts()

1    6691
2     625
3      99
4      36
5      16
6      10
7       6
8       2
Name: nfams, dtype: int64

In [17]:
df['ncouples'].value_counts()

1    4905
0    2467
2     110
3       3
Name: ncouples, dtype: int64

In [18]:
df['nmothers'].value_counts()

1    4469
0    2621
2     379
3      16
Name: nmothers, dtype: int64

In [19]:
df['nfathers'].value_counts()

1    3911
0    3465
2     108
3       1
Name: nfathers, dtype: int64

In [20]:
df['momloc'].value_counts()

0     4913
2     1840
1      607
3       59
5       23
4       17
6       16
8        6
7        2
10       1
9        1
Name: momloc, dtype: int64

In [21]:
df['stepmom'].value_counts()

0    7378
7      89
4       8
2       6
1       4
Name: stepmom, dtype: int64

In [22]:
df['momrule'].value_counts()

0    4913
1    2491
2      36
3      20
5      16
7       6
6       3
Name: momrule, dtype: int64

In [23]:
df['poploc'].value_counts()

0    5504
1    1872
2      82
5       7
4       7
3       6
6       3
7       2
9       1
8       1
Name: poploc, dtype: int64

In [24]:
df['steppop'].value_counts()

0    7480
1       4
2       1
Name: steppop, dtype: int64

In [25]:
df['poprule'].value_counts()

0    5504
1    1955
2      16
3       9
7       1
Name: poprule, dtype: int64

In [26]:
df['sploc'].value_counts()

0     4581
1     1430
2     1425
4       19
3       14
5        6
6        4
7        3
8        2
10       1
Name: sploc, dtype: int64

In [27]:
df['sprule'].value_counts()

0    4581
1    2734
2     149
3      10
4       8
5       3
Name: sprule, dtype: int64

In [28]:
df['famsize'].value_counts()

2     1469
4     1365
1     1281
3     1269
5      991
6      531
7      320
8      118
9       56
10      32
11      29
13      12
12       9
14       2
21       1
Name: famsize, dtype: int64

In [29]:
df['nchild'].value_counts()

0    5292
1     897
2     704
3     373
4     135
5      47
6      17
7      13
9       5
8       2
Name: nchild, dtype: int64

In [30]:
df['nchlt5'].value_counts()

0    6739
1     547
2     177
3      22
Name: nchlt5, dtype: int64

In [31]:
df['famunit'].value_counts()

1    7152
2     273
3      45
4       9
5       5
7       1
Name: famunit, dtype: int64

In [32]:
df['eldch'].value_counts()

99    5292
16      93
17      93
19      89
18      88
20      87
4       85
3       83
5       82
9       82
10      82
14      81
15      79
11      79
0       79
1       78
8       76
12      76
21      74
13      70
6       68
7       68
2       67
23      60
22      54
24      44
26      30
25      27
27      20
30      18
      ... 
55       5
40       5
51       5
42       5
35       5
50       4
56       4
39       4
49       4
52       4
46       3
61       3
43       3
54       3
53       3
65       2
36       2
63       2
60       2
48       2
57       2
62       2
47       1
44       1
67       1
58       1
59       1
66       1
68       1
64       1
Name: eldch, Length: 69, dtype: int64

In [33]:
df['yngch'].value_counts()

99    5292
0      205
1      164
2      141
3      129
4      107
6      101
5       89
9       88
8       83
7       78
10      73
13      69
16      69
14      68
15      68
12      67
11      62
17      52
18      49
20      48
19      44
21      42
22      33
23      28
24      25
27      18
26      15
25      15
32      12
      ... 
34       6
37       6
51       5
52       5
61       4
49       4
42       4
46       4
50       4
35       4
56       4
53       3
55       3
54       3
36       2
48       2
65       2
60       2
57       2
63       2
62       2
39       2
43       2
44       1
64       1
68       1
58       1
66       1
47       1
59       1
Name: yngch, Length: 68, dtype: int64

In [34]:
df['nsibs'].value_counts()

0    5146
1    1048
2     610
3     334
4     197
5      77
6      37
7      25
8       8
9       3
Name: nsibs, dtype: int64

In [35]:
df['relateg'].value_counts()

Head_Householder          2683
Child                     2485
Spouse                    1429
Partner_friend_visitor     202
Other_non-relatives        190
Sibling                    134
Parent                      79
Grandchild                  76
Institutional_inmates       73
Other_relatives             69
Sibling-in-Law              33
Parent-in-Law               19
Child-in-law                13
Name: relateg, dtype: int64

In [36]:
df['age'].value_counts()

19    164
22    162
23    156
25    154
20    154
26    150
33    148
32    146
30    145
12    140
17    140
24    139
21    137
29    135
18    134
15    127
27    126
0     122
16    122
31    122
9     121
14    120
11    119
28    118
10    118
5     117
13    116
3     111
1     111
39    109
     ... 
52     60
48     57
60     56
68     55
66     55
69     54
67     47
65     43
64     41
71     36
72     34
70     34
73     34
75     33
74     29
76     25
79     24
81     23
83     22
77     19
90     19
82     18
80     18
78     17
85     13
84     11
87      8
89      8
86      7
88      3
Name: age, Length: 91, dtype: int64

In [37]:
df['age'].describe()

count    7485.000000
mean       32.486707
std        21.120395
min         0.000000
25%        16.000000
50%        29.000000
75%        48.000000
max        90.000000
Name: age, dtype: float64

In [38]:
df['sex'].value_counts()

Female    3878
Male      3607
Name: sex, dtype: int64

In [39]:
df['raceg'].value_counts()

White                     5854
Black_Negro                975
Other_Asian_or_Pacific     269
Japanese                   124
Other_race_nec             114
Chinese                     93
American_Indian             56
Name: raceg, dtype: int64

In [40]:
df['marst'].value_counts()

Never_married_single      3425
Married_spouse_present    2904
Divorced                   514
Widowed                    366
Separated                  186
Married_spouse_absent       90
Name: marst, dtype: int64

In [41]:
df['chborn'].value_counts()

No_children           2775
2__children           1449
1_child               1256
3                      884
4                      552
5                      251
6                      123
7                       88
8                       37
9                       27
10_children             18
0                       10
12_(12+_1960-1990)       9
11                       6
Name: chborn, dtype: int64

In [42]:
df['bplg'].value_counts()

California                    3080
Mexico                         710
New_York                       265
Texas                          259
Illinois                       219
Ohio                           141
Pennsylvania                   135
Central_America                115
Missouri                       105
Abroad_(unknown)_or_at_sea      93
Louisiana                       84
Oklahoma                        83
Michigan                        82
Philippines                     82
Iowa                            71
Minnesota                       66
Kansas                          62
Indiana                         61
Colorado                        61
Washington                      60
China                           59
Arkansas                        58
South_America                   58
Nebraska                        57
Korea                           56
Canada                          54
Mississippi                     53
New_Jersey                      53
Massachusetts       

In [43]:
df['school'].value_counts()

No_not_in_school    5314
Yes_in_school       2171
Name: school, dtype: int64

In [44]:
df['educrec'].value_counts()

Grade_12                   1807
1_to_3_years_of_college    1294
Grade_5-8                  1174
4+_years_of_college         894
Grade_1-4                   651
None_or_preschool           571
Grade_11                    425
Grade_10                    347
Grade_9                     322
Name: educrec, dtype: int64

In [45]:
df['schltype'].value_counts()

Public_school            6192
Church-related_(1980)     862
Other_private_1980        426
0                           5
Name: schltype, dtype: int64

In [46]:
df['empstatg'].value_counts()

Employed              4539
Not_in_labor_force    2626
Unemployed             316
0                        4
Name: empstatg, dtype: int64

In [47]:
df['labforce'].value_counts()

Yes_in_labor_force       4855
No_not_in_labor_force    2626
0                           4
Name: labforce, dtype: int64

In [48]:
df['occ1950'].value_counts()

Operative_and_kindred_workers_(nec)                   799
Clerical_and_kindred_workers_(n_e_c_)                 635
Managers_officials_and_proprietors_(nec)              606
Salesmen_and_sales_clerks_(nec)                       381
Stenographers_typists_and_secretaries                 353
Laborers_(nec)                                        249
Teachers_(n_e_c_)                                     171
Foremen_(nec)                                         167
Bookkeepers                                           155
Janitors_and_sextons                                  152
Cashiers                                              141
Cooks_except_private_household                        136
Attendants_hospital_and_other_institution             115
Waiters_and_waitresses                                114
Professional_technical_99_kindred_workers_(nec)        96
Nurses_professional                                    90
Office_machine_operators                               89
Truck_and_trac

In [49]:
df['occscore'].value_counts()

0     3187
22     493
24     470
41     350
23     329
25     272
26     254
20     239
37     133
11     130
31     116
18     113
27     112
17     106
30     103
16     100
35      96
32      79
19      73
29      71
38      62
14      53
34      49
6       47
13      44
33      43
39      42
42      35
21      34
61      33
46      33
28      28
48      25
36      24
79      23
15      17
9       12
49      10
43      10
40       7
10       7
45       7
4        7
50       3
5        2
58       1
63       1
Name: occscore, dtype: int64

In [50]:
df['sei'].value_counts()

0     3187
18     481
44     466
68     362
47     239
61     215
8      161
15     143
72     141
19     119
49     114
16      97
9       88
51      86
84      79
22      74
45      74
52      65
11      59
13      58
65      56
27      52
46      52
62      49
26      46
33      45
78      45
48      42
53      41
7       41
      ... 
58      12
60      11
77       9
90       9
76       8
37       8
50       7
28       7
40       6
81       6
12       5
63       5
69       5
79       4
25       4
4        3
21       3
41       3
20       2
5        2
43       2
56       2
80       2
59       1
96       1
35       1
54       1
75       1
85       1
14       1
Name: sei, Length: 82, dtype: int64

In [51]:
df['ind1950'].value_counts()

Educational_services                                  524
Eating_and_drinking__places                           401
Hospitals                                             395
Construction                                          347
Misc_business_services                                331
Aircraft_and_parts                                    289
General_merchandise                                   204
Banking_and_credit                                    198
Printing_publishing_and_allied_industries             189
Electrical_machinery_equipment_and_supplies           181
Medical_and_other_health_services_except_hospitals    176
Apparel_and_accessories                               158
Federal_public_administration                         151
Food_stores_except_dairy                              149
Insurance                                             137
Welfare_and_religious_services                        132
Misc_machinery                                        127
Auto_repair_se

In [52]:
df['classwkg'].value_counts()

Works_for_wages_salary    6856
Self-employed              624
0                            5
Name: classwkg, dtype: int64

In [53]:
df['wkswork2'].value_counts()

50-52_weeks    4424
40-47_weeks     744
14-26_weeks     649
1-13_weeks      603
27-39_weeks     553
48-49_weeks     507
0                 5
Name: wkswork2, dtype: int64

In [54]:
df['hrswork2'].value_counts()

40_hours       3971
15-29_hours     753
41-48_hours     719
49-59_hours     679
1-14_hours      392
30-34_hours     346
60+_hours       322
35-39_hours     298
0                 5
Name: hrswork2, dtype: int64

In [55]:
df['yrlastwk'].value_counts()

Worked_more_than_10_yrs_ago    1768
Never_worked                   1707
Worked_previous_yr             1012
Worked_current_yr               966
Worked_6-10_yrs_ago             925
Worked_3-5_yrs_ago              710
Worked_2_yrs_prior              388
0                                 9
Name: yrlastwk, dtype: int64

In [56]:
df['workedyr'].value_counts()

Yes    4989
No     2492
0         4
Name: workedyr, dtype: int64

In [57]:
df['inctot'].value_counts()

 999999    1684
 0         1036
 10005      101
 8005        76
 5005        75
 6005        74
 12005       71
 7005        63
 15005       60
 3005        59
 20005       57
 4005        54
 18005       49
 11005       49
 2005        45
 16005       45
 9005        44
 13005       36
 75000       33
 14005       33
 30005       25
 605         25
 17005       25
 2505        25
 22005       24
 3605        23
 25005       23
 19005       23
 1205        22
 1505        22
           ... 
 1030         1
 25610        1
 11275        1
 15385        1
 810          1
 8990         1
 27415        1
 15125        1
 21140        1
 2715         1
 6825         1
 74410        1
 31405        1
 15025        1
 27315        1
 10935        1
-835          1
 15045        1
 2775         1
 15065        1
 13020        1
 10995        1
 23285        1
 4860         1
 29440        1
 15105        1
 770          1
 23305        1
 2835         1
 26525        1
Name: inctot, Length: 20

In [58]:
df['incwage'].value_counts()

0         2081
999999    1811
10005      112
12005       88
8005        85
6005        78
20005       77
5005        73
15005       71
7005        71
18005       66
16005       65
13005       60
11005       56
3005        55
4005        55
2005        49
9005        49
14005       44
25005       43
17005       38
22005       34
19005       31
30005       30
605         29
1505        28
21005       25
24005       24
1005        23
1205        22
          ... 
12925        1
5495         1
9585         1
42345        1
17765        1
21855        1
9565         1
1365         1
13795        1
3535         1
34285        1
17905        1
13975        1
7625         1
30335        1
59005        1
34405        1
1625         1
535          1
12825        1
25115        1
37405        1
5695         1
5675         1
555          1
25135        1
19915        1
9725         1
32205        1
28475        1
Name: incwage, Length: 1049, dtype: int64

In [59]:
df['incfarm'].value_counts()

 0         5656
 999999    1811
 805          2
 2005         1
 3005         1
 61005        1
 505          1
 705          1
 1005         1
 10005        1
-1505         1
 1255         1
 3755         1
-4005         1
 55           1
 75           1
 435          1
 455          1
 2755         1
Name: incfarm, dtype: int64

In [60]:
df['incbus'].value_counts()

 0         5352
 999999    1811
 30005       12
 5005        11
 15005       10
 3005         9
 8005         9
 10005        8
 6005         7
 20005        7
 505          7
 75000        6
 40005        5
 25005        5
 2005         5
 50005        5
 7005         5
 3505         5
 4005         5
 18005        4
 1755         4
 24005        3
 14005        3
 16005        3
 35005        3
 6255         3
 5255         3
 1005         3
-4005         3
 10505        3
           ... 
 12305        1
 255          1
 22865        1
-2385         1
-1555         1
 12805        1
 42005        1
-505          1
 1205         1
-1235         1
-1255         1
 1305         1
 9505         1
 1405         1
 9605         1
 5535         1
 1605         1
-2325         1
 435          1
-1755         1
 1805         1
 415          1
-6005         1
 26505        1
 6055         1
-345          1
 6135         1
 14505        1
 12605        1
 25505        1
Name: incbus, Length: 16

In [61]:
df['incss'].value_counts()

0        5060
99999    1684
3605       20
3005       15
2405       13
4805       13
5005        9
2505        9
4005        8
6005        8
4205        8
1205        7
4325        7
3575        7
3505        7
1855        6
2005        6
1655        6
1955        6
4085        6
3655        6
4055        5
1805        5
3555        5
3055        5
4645        4
5045        4
1745        4
1605        4
5505        4
         ... 
4555        1
595         1
405         1
4495        1
4485        1
4955        1
865         1
905         1
5615        1
5595        1
1485        1
1465        1
5555        1
1425        1
175         1
1385        1
5475        1
4285        1
1285        1
1265        1
215         1
1225        1
5255        1
255         1
4385        1
1085        1
1045        1
335         1
5075        1
2015        1
Name: incss, Length: 362, dtype: int64

In [62]:
df['incwelfr'].value_counts()

0        5455
99999    1684
4925        9
3005        9
355         7
4005        7
4275        6
3605        6
2455        5
4205        5
1255        5
255         4
3965        4
155         4
3955        4
805         4
1005        4
3975        4
3685        3
4055        3
1835        3
4285        3
3455        3
2505        3
6005        3
1445        3
5505        3
5845        3
605         3
555         3
         ... 
8235        1
9005        1
845         1
4955        1
965         1
755         1
1085        1
1105        1
655         1
1265        1
1285        1
4665        1
9525        1
4545        1
2155        1
1505        1
1525        1
1565        1
1665        1
1685        1
5795        1
1725        1
5835        1
5855        1
1805        1
1825        1
1845        1
2045        1
4465        1
2005        1
Name: incwelfr, Length: 207, dtype: int64

In [63]:
df['incother'].value_counts()

0        5227
99999    1684
605        12
2005       10
1205       10
4005        9
1805        9
1005        8
2505        8
505         8
3605        7
12005       7
205         7
305         7
5005        7
1505        7
3255        6
10005       6
805         6
905         6
485         6
6005        6
3005        6
965         5
1055        5
65          5
2755        5
355         5
555         5
655         5
         ... 
665         1
21005       1
645         1
8805        1
3895        1
8745        1
395         1
4385        1
6105        1
1165        1
1685        1
1665        1
1645        1
175         1
4285        1
195         1
1465        1
1445        1
235         1
1345        1
1325        1
5415        1
9505        1
275         1
1285        1
5165        1
1245        1
1225        1
9755        1
6125        1
Name: incother, Length: 283, dtype: int64

In [64]:
df['poverty'].value_counts()

501    1676
1       164
0       132
79       39
211      35
203      35
185      33
119      31
106      31
316      31
264      29
325      27
207      27
102      26
406      26
352      26
159      26
237      25
287      25
395      24
246      24
108      23
132      23
222      23
299      23
230      23
153      23
348      23
189      23
172      22
       ... 
4         3
12        3
407       3
440       3
436       3
28        3
20        3
5         3
397       3
17        3
31        3
357       3
469       3
382       3
244       3
49        3
45        3
25        3
413       3
470       2
464       2
475       2
11        2
44        1
486       1
42        1
2         1
8         1
15        1
499       1
Name: poverty, Length: 501, dtype: int64

In [65]:
df['migrat5g'].value_counts()

Same_house                          3517
Same_state_countydifferent_house    2586
Abroad                               541
Different_state                      492
Same_state_different_county          346
0                                      3
Name: migrat5g, dtype: int64

In [66]:
df['movedin'].value_counts() ## target variable

0    4802
2     860
1     726
5     412
6     401
7     213
8      71
Name: movedin, dtype: int64

In [67]:
df['vetstat'].value_counts()

No_Service    6318
Yes           1163
0                4
Name: vetstat, dtype: int64

In [68]:
df['tranwork'].value_counts()

Auto                5715
Bus_or_streetcar     499
Truck                484
Walked_only          331
Van                  185
Worked_at_home       109
Motorcycle            71
Bicycle               45
Other                 40
0                      5
Taxicab                1
Name: tranwork, dtype: int64

In [69]:
# INSERTING THE DATA INTO NEW NAME
ip=df
ip.head(5)

year                                                gq         gqtypeg  \
0    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
1    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
2    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
3    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
4    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   

       farm                      ownershg   value  rent  ftotinc  nfams  \
0  Non-Farm                        Rented  999999   325     9505      1   
1  Non-Farm                        Rented  999999   287     8005      1   
2  Non-Farm  Owned_or_being_bought_(loan)   85000     0    29635      1   
3  Non-Farm  Owned_or_being_bought_(loan)  999999     0    52895      1   
4  Non-Farm                        Rented  999999   185     7610      2   

   ncouples  ...  incfarm  incss  incwelfr  incother  poverty  \
0         1  ...        0      0         0         0      128   
1         0  ...        0      0         0         0      211   
2         1  ...   999999  99999     99999     99999      304   
3         1  ...        0      0         0         0      501   
4         0  ...        0      0         0         0      119   

                           migrat5g    migplac5  movedin     vetstat  \
0  Same_state_countydifferent_house  California        0  No_Service   
1  Same_state_countydifferent_house  California        1  No_Service   
2                        Same_house  Same_house        0  No_Service   
3                        Same_house  Same_house        0  No_Service   
4  Same_state_countydifferent_house  California        0  No_Service   

           tranwork  
0  Bus_or_streetcar  
1       Walked_only  
2              Auto  
3       Walked_only  
4              Auto  

[5 rows x 61 columns]

#### CONVERTING THE TWO CATEGORIES AS "0" AND "1"
#### FOR EXAMPLE:- FEMALE AS "1" AND MALE AS "0"

In [70]:
ip['sex']=[1 if x=='Female' else 0 for x in ip['sex']]
ip['sex'].value_counts()

1    3878
0    3607
Name: sex, dtype: int64

In [71]:
ip['school']=[1 if x=='Yes_in_school' else 0 for x in ip['school']]
ip['school'].value_counts()

0    5314
1    2171
Name: school, dtype: int64

In [72]:
ip['farm']=[1 if x=='Farm' else 0 for x in ip['farm']]
ip['farm'].value_counts()

0    7482
1       3
Name: farm, dtype: int64

In [73]:
ip['ownershg']=[1 if x=='Rented' else 0 for x in ip['ownershg']]
ip['ownershg'].value_counts()

0    4169
1    3316
Name: ownershg, dtype: int64

In [74]:
ip.head(5)

year                                                gq         gqtypeg  \
0    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
1    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
2    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
3    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
4    98  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   

   farm  ownershg   value  rent  ftotinc  nfams  ncouples  ...  incfarm  \
0     0         1  999999   325     9505      1         1  ...        0   
1     0         1  999999   287     8005      1         0  ...        0   
2     0         0   85000     0    29635      1         1  ...   999999   
3     0         0  999999     0    52895      1         1  ...        0   
4     0         1  999999   185     7610      2         0  ...        0   

   incss  incwelfr  incother  poverty                          migrat5g  \
0      0         0         0      128  Same_state_countydifferent_house   
1      0         0         0      211  Same_state_countydifferent_house   
2  99999     99999     99999      304                        Same_house   
3      0         0         0      501                        Same_house   
4      0         0         0      119  Same_state_countydifferent_house   

     migplac5  movedin     vetstat          tranwork  
0  California        0  No_Service  Bus_or_streetcar  
1  California        1  No_Service       Walked_only  
2  Same_house        0  No_Service              Auto  
3  Same_house        0  No_Service       Walked_only  
4  California        0  No_Service              Auto  

[5 rows x 61 columns]

In [75]:
#CHECKING THE NO OF NUMERICAL AND CATEGORICAL COLUMNS IN THE DATASET
ip.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7485 entries, 0 to 7484
Data columns (total 61 columns):
year        7485 non-null int64
gq          7485 non-null object
gqtypeg     7485 non-null object
farm        7485 non-null int64
ownershg    7485 non-null int64
value       7485 non-null int64
rent        7485 non-null int64
ftotinc     7485 non-null int64
nfams       7485 non-null int64
ncouples    7485 non-null int64
nmothers    7485 non-null int64
nfathers    7485 non-null int64
momloc      7485 non-null int64
stepmom     7485 non-null int64
momrule     7485 non-null int64
poploc      7485 non-null int64
steppop     7485 non-null int64
poprule     7485 non-null int64
sploc       7485 non-null int64
sprule      7485 non-null int64
famsize     7485 non-null int64
nchild      7485 non-null int64
nchlt5      7485 non-null int64
famunit     7485 non-null int64
eldch       7485 non-null int64
yngch       7485 non-null int64
nsibs       7485 non-null int64
relateg     7485 non-null o

### ALL NUMERICAL COLUMNS

In [76]:
#PASSING ALL THE NUMERICAL COLUMNS INTO A LIST
List=['year','farm','ownershg','value','rent','ftotinc','nfams','ncouples','nmothers','nfathers','momloc','stepmom','momrule','poploc',
    'steppop','poprule','sploc','sprule','famsize','nchild','nchlt5','famunit','eldch','yngch','nsibs','age','sex','school','occscore'
    'sei','incwage','inctot','incfarm','incbus','incss','incwelfr','incother','poverty']

In [77]:
#GIVING ALL THE NUMERICAL COLUMNS
ip1=ip.loc[:, List]

C:\ProgramData\Anaconda3\lib\site-packages\pandas\core\indexing.py:1494: FutureWarning: 
Passing list-likes to .loc or [] with any missing label will raise
KeyError in the future, you can use .reindex() as an alternative.

See the documentation here:
https://pandas.pydata.org/pandas-docs/stable/indexing.html#deprecate-loc-reindex-listlike
  return self._getitem_tuple(key)


In [78]:
ip1.head()

year  farm  ownershg   value  rent  ftotinc  nfams  ncouples  nmothers  \
0    98     0         1  999999   325     9505      1         1         1   
1    98     0         1  999999   287     8005      1         0         0   
2    98     0         0   85000     0    29635      1         1         1   
3    98     0         0  999999     0    52895      1         1         1   
4    98     0         1  999999   185     7610      2         0         0   

   nfathers  ...  school  occscoresei  incwage  inctot  incfarm  incbus  \
0         1  ...       0          NaN        0       0        0       0   
1         0  ...       0          NaN     8005    8005        0       0   
2         1  ...       1          NaN   999999  999999   999999  999999   
3         1  ...       1          NaN        0       0        0       0   
4         0  ...       1          NaN     4505    4520        0       0   

   incss  incwelfr  incother  poverty  
0      0         0         0      128  
1      0         0         0      211  
2  99999     99999     99999      304  
3      0         0         0      501  
4      0         0         0      119  

[5 rows x 37 columns]

In [79]:
#CHECKING THE COLUMNS OF NUMERICAL COLUMNS
ip1.columns

Index(['year', 'farm', 'ownershg', 'value', 'rent', 'ftotinc', 'nfams',
       'ncouples', 'nmothers', 'nfathers', 'momloc', 'stepmom', 'momrule',
       'poploc', 'steppop', 'poprule', 'sploc', 'sprule', 'famsize', 'nchild',
       'nchlt5', 'famunit', 'eldch', 'yngch', 'nsibs', 'age', 'sex', 'school',
       'occscoresei', 'incwage', 'inctot', 'incfarm', 'incbus', 'incss',
       'incwelfr', 'incother', 'poverty'],
      dtype='object')

In [80]:
#THERE ARE 37 NUMERICAL COLUMNS 
ip1.shape

(7485, 37)

### NORMALIZE THE NUMERICAL COLUMNS

In [81]:
# WE NORMALIZE THE VALUES OF ALL NUMERICAL COLUMNSINTO 0-1
import sklearn
from sklearn import preprocessing
min_max_scaler = preprocessing.MinMaxScaler()
x_scaled = min_max_scaler.fit_transform(ip1)

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\preprocessing\data.py:334: DataConversionWarning: Data with input dtype int64, float64 were all converted to float64 by MinMaxScaler.
  return self.partial_fit(X, y)
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\preprocessing\data.py:364: RuntimeWarning: All-NaN slice encountered
  data_min = np.nanmin(X, axis=0)
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\preprocessing\data.py:365: RuntimeWarning: All-NaN slice encountered
  data_max = np.nanmax(X, axis=0)


In [82]:
#CONVERTING INTO DATAFRAME
ip_normalized = pd.DataFrame(x_scaled,columns=['year', 'farm', 'ownershg', 'value', 'rent', 'ftotinc', 'nfams',
       'ncouples', 'nmothers', 'nfathers', 'momloc', 'stepmom', 'momrule',
       'poploc', 'steppop', 'poprule', 'sploc', 'sprule', 'famsize', 'nchild',
       'nchlt5', 'famunit', 'eldch', 'yngch', 'nsibs', 'age', 'sex', 'school',
       'occscoresei', 'incwage', 'inctot', 'incfarm', 'incbus', 'incss',
       'incwelfr', 'incother', 'poverty'])

In [83]:
ip_normalized.head() #normalized dataframe

year  farm  ownershg     value   rent   ftotinc     nfams  ncouples  \
0   0.0   0.0       1.0  1.000000  0.650  0.018549  0.000000  0.333333   
1   0.0   0.0       1.0  1.000000  0.574  0.017063  0.000000  0.000000   
2   0.0   0.0       0.0  0.080402  0.000  0.038495  0.000000  0.333333   
3   0.0   0.0       0.0  1.000000  0.000  0.061543  0.000000  0.333333   
4   0.0   0.0       1.0  1.000000  0.370  0.016671  0.142857  0.000000   

   nmothers  nfathers  ...  school  occscoresei   incwage    inctot   incfarm  \
0  0.333333  0.333333  ...     0.0          NaN  0.000000  0.009131  0.003989   
1  0.000000  0.000000  ...     0.0          NaN  0.008005  0.017063  0.003989   
2  0.333333  0.333333  ...     1.0          NaN  1.000000  1.000000  1.000000   
3  0.333333  0.333333  ...     1.0          NaN  0.000000  0.009131  0.003989   
4  0.000000  0.000000  ...     1.0          NaN  0.004505  0.013610  0.003989   

     incbus  incss  incwelfr  incother   poverty  
0  0.008925    0.0       0.0       0.0  0.255489  
1  0.008925    0.0       0.0       0.0  0.421158  
2  1.000000    1.0       1.0       1.0  0.606786  
3  0.008925    0.0       0.0       0.0  1.000000  
4  0.008925    0.0       0.0       0.0  0.237525  

[5 rows x 37 columns]

In [84]:
ip_normalized.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7485 entries, 0 to 7484
Data columns (total 37 columns):
year           7485 non-null float64
farm           7485 non-null float64
ownershg       7485 non-null float64
value          7485 non-null float64
rent           7485 non-null float64
ftotinc        7485 non-null float64
nfams          7485 non-null float64
ncouples       7485 non-null float64
nmothers       7485 non-null float64
nfathers       7485 non-null float64
momloc         7485 non-null float64
stepmom        7485 non-null float64
momrule        7485 non-null float64
poploc         7485 non-null float64
steppop        7485 non-null float64
poprule        7485 non-null float64
sploc          7485 non-null float64
sprule         7485 non-null float64
famsize        7485 non-null float64
nchild         7485 non-null float64
nchlt5         7485 non-null float64
famunit        7485 non-null float64
eldch          7485 non-null float64
yngch          7485 non-null float64
nsibs

In [85]:
# THERE IS NO VALUES IN "occscoresei" ONLY "NAN" 
ip_normalized['occscoresei'].value_counts()

Series([], Name: occscoresei, dtype: int64)

In [86]:
ip_normalized.columns

Index(['year', 'farm', 'ownershg', 'value', 'rent', 'ftotinc', 'nfams',
       'ncouples', 'nmothers', 'nfathers', 'momloc', 'stepmom', 'momrule',
       'poploc', 'steppop', 'poprule', 'sploc', 'sprule', 'famsize', 'nchild',
       'nchlt5', 'famunit', 'eldch', 'yngch', 'nsibs', 'age', 'sex', 'school',
       'occscoresei', 'incwage', 'inctot', 'incfarm', 'incbus', 'incss',
       'incwelfr', 'incother', 'poverty'],
      dtype='object')

In [87]:
#DDROPPING THE COLUMN
ip_normalized=ip_normalized.drop(columns=['occscoresei'])
ip_normalized.head()

year  farm  ownershg     value   rent   ftotinc     nfams  ncouples  \
0   0.0   0.0       1.0  1.000000  0.650  0.018549  0.000000  0.333333   
1   0.0   0.0       1.0  1.000000  0.574  0.017063  0.000000  0.000000   
2   0.0   0.0       0.0  0.080402  0.000  0.038495  0.000000  0.333333   
3   0.0   0.0       0.0  1.000000  0.000  0.061543  0.000000  0.333333   
4   0.0   0.0       1.0  1.000000  0.370  0.016671  0.142857  0.000000   

   nmothers  nfathers  ...  sex  school   incwage    inctot   incfarm  \
0  0.333333  0.333333  ...  1.0     0.0  0.000000  0.009131  0.003989   
1  0.000000  0.000000  ...  1.0     0.0  0.008005  0.017063  0.003989   
2  0.333333  0.333333  ...  1.0     1.0  1.000000  1.000000  1.000000   
3  0.333333  0.333333  ...  0.0     1.0  0.000000  0.009131  0.003989   
4  0.000000  0.000000  ...  1.0     1.0  0.004505  0.013610  0.003989   

     incbus  incss  incwelfr  incother   poverty  
0  0.008925    0.0       0.0       0.0  0.255489  
1  0.008925    0.0       0.0       0.0  0.421158  
2  1.000000    1.0       1.0       1.0  0.606786  
3  0.008925    0.0       0.0       0.0  1.000000  
4  0.008925    0.0       0.0       0.0  0.237525  

[5 rows x 36 columns]

In [88]:
#AFTER DROPPING THE COLUMNNAMES
ip_normalized.columns

Index(['year', 'farm', 'ownershg', 'value', 'rent', 'ftotinc', 'nfams',
       'ncouples', 'nmothers', 'nfathers', 'momloc', 'stepmom', 'momrule',
       'poploc', 'steppop', 'poprule', 'sploc', 'sprule', 'famsize', 'nchild',
       'nchlt5', 'famunit', 'eldch', 'yngch', 'nsibs', 'age', 'sex', 'school',
       'incwage', 'inctot', 'incfarm', 'incbus', 'incss', 'incwelfr',
       'incother', 'poverty'],
      dtype='object')

### ALL CATEGORICAL COLUMNS

In [89]:
# limit to categorical data using df.select_dtypes()
# Taking all the object datatypes as categorical columns without giving names
X = ip.select_dtypes(include=[object])
X.head(3)

gq         gqtypeg  \
0  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
1  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   
2  HH_in_1970_definition_but_sampled_as_larger_unit  NA_(household)   

            relateg        raceg                   marst       chborn  \
0            Spouse        White  Married_spouse_present  2__children   
1  Head_Householder  Black_Negro    Never_married_single  No_children   
2             Child        White    Never_married_single  No_children   

         bplg                  educrec               schltype  \
0  California  1_to_3_years_of_college          Public_school   
1  California                 Grade_12  Church-related_(1980)   
2  California        None_or_preschool     Other_private_1980   

             empstatg  ...                                     ind1950  \
0  Not_in_labor_force  ...                    Food_stores_except_dairy   
1            Employed  ...  Crude_petroleum_and_natural_gas_extraction   
2  Not_in_labor_force  ...                 Eating_and_drinking__places   

                 classwkg     wkswork2     hrswork2            yrlastwk  \
0  Works_for_wages_salary  14-26_weeks     40_hours  Worked_3-5_yrs_ago   
1  Works_for_wages_salary  40-47_weeks     40_hours   Worked_current_yr   
2  Works_for_wages_salary  50-52_weeks  30-34_hours        Never_worked   

  workedyr                          migrat5g    migplac5     vetstat  \
0       No  Same_state_countydifferent_house  California  No_Service   
1      Yes  Same_state_countydifferent_house  California  No_Service   
2       No                        Same_house  Same_house  No_Service   

           tranwork  
0  Bus_or_streetcar  
1       Walked_only  
2              Auto  

[3 rows x 22 columns]

In [90]:
# check original shape of categorical columns
X.shape

(7485, 22)

In [91]:
# view the categorical columns using df.columns
X.columns

Index(['gq', 'gqtypeg', 'relateg', 'raceg', 'marst', 'chborn', 'bplg',
       'educrec', 'schltype', 'empstatg', 'labforce', 'occ1950', 'ind1950',
       'classwkg', 'wkswork2', 'hrswork2', 'yrlastwk', 'workedyr', 'migrat5g',
       'migplac5', 'vetstat', 'tranwork'],
      dtype='object')

### LABEL-ENCODING

In [92]:
# import preprocessing from sklearn
from sklearn import preprocessing

In [93]:
#create a LabelEncoder object and fit it to each feature in X
# encode labels with value between 0 and n_classes-1.
le = preprocessing.LabelEncoder()
# used df.apply() to apply le.fit_transform to all columns
X_2 = X.apply(le.fit_transform)
X_2.head()

gq  gqtypeg  relateg  raceg  marst  chborn  bplg  educrec  schltype  \
0   0        4       12      6      2       5    10        0         3   
1   0        4        3      1      3      13    10        5         1   
2   0        4        0      6      3      13    10        8         2   
3   0        4        0      6      3       4    10        3         3   
4   0        4        9      6      3      13    10        0         3   

   empstatg  ...  ind1950  classwkg  wkswork2  hrswork2  yrlastwk  workedyr  \
0         2  ...       39         2         2         5         3         1   
1         1  ...       20         2         4         5         5         2   
2         2  ...       28         2         6         3         1         1   
3         2  ...       50         2         4         7         1         1   
4         1  ...       50         2         4         2         6         2   

   migrat5g  migplac5  vetstat  tranwork  
0         4         5        1         3  
1         4         5        1         9  
2         3        39        1         1  
3         3        39        1         9  
4         4         5        1         1  

[5 rows x 22 columns]

In [94]:
#checking the categorical columns
X_2.shape

(7485, 22)

In [95]:
#successfully conveted the objects into integers
X_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7485 entries, 0 to 7484
Data columns (total 22 columns):
gq          7485 non-null int32
gqtypeg     7485 non-null int32
relateg     7485 non-null int32
raceg       7485 non-null int32
marst       7485 non-null int32
chborn      7485 non-null int32
bplg        7485 non-null int32
educrec     7485 non-null int32
schltype    7485 non-null int32
empstatg    7485 non-null int32
labforce    7485 non-null int32
occ1950     7485 non-null int32
ind1950     7485 non-null int32
classwkg    7485 non-null int32
wkswork2    7485 non-null int32
hrswork2    7485 non-null int32
yrlastwk    7485 non-null int32
workedyr    7485 non-null int32
migrat5g    7485 non-null int32
migplac5    7485 non-null int32
vetstat     7485 non-null int32
tranwork    7485 non-null int32
dtypes: int32(22)
memory usage: 643.3 KB


### ONE-HOT-ENCODING 
A one hot encoding is a representation of categorical variables as binary vectors.

This first requires that the categorical values be mapped to integer values.

Then, each integer value is represented as a binary vector that is all zero values except the index of the integer, which is marked with a 1.

In [96]:
# create a OneHotEncoder object, and fit it to all of X

# 1. INSTANTIATE
enc = preprocessing.OneHotEncoder()

# 2. FIT
enc.fit(X_2)

# 3. Transform
onehotlabels = enc.transform(X_2).toarray()
onehotlabels.shape

# as you can see, you've the same number of rows 595
# but now you've so many more columns due to how we changed all the categorical data into numerical data

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\preprocessing\_encoders.py:371: FutureWarning: The handling of integer data will change in version 0.22. Currently, the categories are determined based on the range [0, max(values)], while in the future they will be determined based on the unique values.
If you want the future behaviour and silence this warning, you can specify "categories='auto'".
In case you used a LabelEncoder before this OneHotEncoder to convert the categories to integers, then you can now use the OneHotEncoder directly.
  warnings.warn(msg, FutureWarning)


(7485, 595)

In [97]:
# converting into dataframe
data = pd.DataFrame(onehotlabels)

In [98]:
#combining all the normalized data and the onehotcoding data 
data[['year', 'farm', 'ownershg', 'value', 'rent', 'ftotinc', 'nfams',
       'ncouples', 'nmothers', 'nfathers', 'momloc', 'stepmom', 'momrule',
       'poploc', 'steppop', 'poprule', 'sploc', 'sprule', 'famsize', 'nchild',
       'nchlt5', 'famunit', 'eldch', 'yngch', 'nsibs', 'age', 'sex', 'school',
       'incwage', 'inctot', 'incfarm', 'incbus', 'incss', 'incwelfr',
       'incother', 'poverty']]=ip_normalized

# DIFFERENT CLASSIFICATION METHODS 

In [99]:
# DIVIDING THE DATA INTO INPUT AND OUTPUT
X = data  # INPUT VARAIABLES
y = df['movedin'] #TARGET VARIAABLES

In [100]:
#CONVERTING THE DATA INTO TEST AND TRAIN
from sklearn.model_selection import train_test_split 
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, 
                                                    random_state=1)

## LogisticRegression

In [101]:
from sklearn.linear_model import LogisticRegression 
from sklearn.metrics import accuracy_score 
model = LogisticRegression()


In [102]:
model.fit(X_train, y_train)

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:460: FutureWarning: Default multi_class will be changed to 'auto' in 0.22. Specify the multi_class option to silence this warning.
  "this warning.", FutureWarning)


LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
          intercept_scaling=1, max_iter=100, multi_class='warn',
          n_jobs=None, penalty='l2', random_state=None, solver='warn',
          tol=0.0001, verbose=0, warm_start=False)

In [103]:
y_pred = model.predict(X_test)

In [104]:
accuracy=accuracy_score(y_test,y_pred)
print(accuracy*100)

77.82231128924516


#### THE ACCURACY OF THE DATA BY USING LOGISTIC REGRESSION  IS 77.8%

In [105]:
from sklearn.metrics import confusion_matrix
cnf_matrix = confusion_matrix(y_test, y_pred)
cnf_matrix

array([[960,   0,   0,   0,   0,   0,   0],
       [  0,  65,  57,   6,   6,   1,   0],
       [  0,  53,  83,  13,   8,   3,   1],
       [  0,  13,  40,  20,  13,   6,   2],
       [  0,   4,  26,  24,  30,  11,   0],
       [  0,   1,  11,   6,  10,   7,   1],
       [  0,   1,   4,   3,   6,   2,   0]], dtype=int64)

## Neural_Network

In [106]:
from sklearn.neural_network import MLPClassifier
from sklearn.metrics import accuracy_score

In [107]:
model = MLPClassifier(hidden_layer_sizes=(2000,1000,800), max_iter=100)
model.fit(X_train, y_train)

MLPClassifier(activation='relu', alpha=0.0001, batch_size='auto', beta_1=0.9,
       beta_2=0.999, early_stopping=False, epsilon=1e-08,
       hidden_layer_sizes=(2000, 1000, 800), learning_rate='constant',
       learning_rate_init=0.001, max_iter=100, momentum=0.9,
       n_iter_no_change=10, nesterovs_momentum=True, power_t=0.5,
       random_state=None, shuffle=True, solver='adam', tol=0.0001,
       validation_fraction=0.1, verbose=False, warm_start=False)

In [108]:
y_pred= model.predict(X_test)

In [109]:
accuracy = model.score(X_test,y_test)
print(accuracy)

0.7668670674682698


#### THE ACCURACY OF THE DATA BY USING NEURAL NETWORKS IS 77.1%

In [110]:
from sklearn.metrics import classification_report, confusion_matrix
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))

[[960   0   0   0   0   0   0]
 [  0  55  56  13   8   3   0]
 [  0  53  76  20   9   2   1]
 [  0  13  33  26  15   4   3]
 [  0   3  24  29  24  11   4]
 [  1   2   7   5  15   5   1]
 [  0   1   1   2   8   2   2]]
              precision    recall  f1-score   support

           0       1.00      1.00      1.00       960
           1       0.43      0.41      0.42       135
           2       0.39      0.47      0.42       161
           5       0.27      0.28      0.28        94
           6       0.30      0.25      0.28        95
           7       0.19      0.14      0.16        36
           8       0.18      0.12      0.15        16

   micro avg       0.77      0.77      0.77      1497
   macro avg       0.39      0.38      0.39      1497
weighted avg       0.76      0.77      0.76      1497



In [111]:
model = MLPClassifier(hidden_layer_sizes=(1500,900,700), max_iter=100)
model.fit(X_train, y_train)

MLPClassifier(activation='relu', alpha=0.0001, batch_size='auto', beta_1=0.9,
       beta_2=0.999, early_stopping=False, epsilon=1e-08,
       hidden_layer_sizes=(1500, 900, 700), learning_rate='constant',
       learning_rate_init=0.001, max_iter=100, momentum=0.9,
       n_iter_no_change=10, nesterovs_momentum=True, power_t=0.5,
       random_state=None, shuffle=True, solver='adam', tol=0.0001,
       validation_fraction=0.1, verbose=False, warm_start=False)

In [112]:
y_pred= model.predict(X_test)

In [113]:
accuracy = model.score(X_test,y_test)
print(accuracy)

0.7755511022044088


#### THE ACCURACY OF THE DATA BY USING NEURAL NETWORKS IS 77.3%

In [114]:
from sklearn.metrics import classification_report, confusion_matrix
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))

[[959   1   0   0   0   0   0]
 [  0  61  53  13   5   3   0]
 [  0  52  75  25   6   2   1]
 [  0  10  28  30  20   4   2]
 [  0   4  22  29  27   8   5]
 [  0   4   6   6  11   8   1]
 [  0   0   3   3   7   2   1]]
              precision    recall  f1-score   support

           0       1.00      1.00      1.00       960
           1       0.46      0.45      0.46       135
           2       0.40      0.47      0.43       161
           5       0.28      0.32      0.30        94
           6       0.36      0.28      0.32        95
           7       0.30      0.22      0.25        36
           8       0.10      0.06      0.08        16

   micro avg       0.78      0.78      0.78      1497
   macro avg       0.41      0.40      0.40      1497
weighted avg       0.77      0.78      0.77      1497



## DECISION TREE

In [115]:
from sklearn.tree import DecisionTreeClassifier # Import Decision Tree Classifier
from sklearn import metrics
# Create Decision Tree classifer object
clf = DecisionTreeClassifier()

# Train Decision Tree Classifer
clf = clf.fit(X_train,y_train)

#Predict the response for test dataset
y_pred = clf.predict(X_test)

In [116]:
# Model Accuracy
print("Accuracy:",metrics.accuracy_score(y_test, y_pred))

Accuracy: 0.770875083500334


#### THE ACCURACY OF THE DATA BY USING DECISION TREE IS 77.6%

In [117]:
from sklearn.metrics import classification_report, confusion_matrix
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))

[[960   0   0   0   0   0   0]
 [  0  68  53   8   3   3   0]
 [  0  58  63  23  10   5   2]
 [  0  18  31  19  17   8   1]
 [  0   3  12  29  30  16   5]
 [  0   2   9   3   7  13   2]
 [  0   2   1   2   7   3   1]]
              precision    recall  f1-score   support

           0       1.00      1.00      1.00       960
           1       0.45      0.50      0.48       135
           2       0.37      0.39      0.38       161
           5       0.23      0.20      0.21        94
           6       0.41      0.32      0.36        95
           7       0.27      0.36      0.31        36
           8       0.09      0.06      0.07        16

   micro avg       0.77      0.77      0.77      1497
   macro avg       0.40      0.41      0.40      1497
weighted avg       0.77      0.77      0.77      1497



## Support Vector Machine

In [118]:
from sklearn.svm import SVC
from sklearn.metrics import accuracy_score 

#model = LogisticRegression()
model = SVC()

model.fit(X, y)

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\svm\base.py:196: FutureWarning: The default value of gamma will change from 'auto' to 'scale' in version 0.22 to account better for unscaled features. Set gamma explicitly to 'auto' or 'scale' to avoid this warning.
  "avoid this warning.", FutureWarning)


SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
  decision_function_shape='ovr', degree=3, gamma='auto_deprecated',
  kernel='rbf', max_iter=-1, probability=False, random_state=None,
  shrinking=True, tol=0.001, verbose=False)

In [119]:
predicted_classes = model.predict(X)

In [120]:
accuracy = accuracy_score(y.values.flatten(),predicted_classes)
print(accuracy)

0.7688710754843019


#### THE ACCURACY OF THE DATA BY USING SUPPORT VECTOR MACHINE IS 76.8%

In [121]:
from sklearn.metrics import classification_report, confusion_matrix
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))

[[960   0   0   0   0   0   0]
 [  0  68  53   8   3   3   0]
 [  0  58  63  23  10   5   2]
 [  0  18  31  19  17   8   1]
 [  0   3  12  29  30  16   5]
 [  0   2   9   3   7  13   2]
 [  0   2   1   2   7   3   1]]
              precision    recall  f1-score   support

           0       1.00      1.00      1.00       960
           1       0.45      0.50      0.48       135
           2       0.37      0.39      0.38       161
           5       0.23      0.20      0.21        94
           6       0.41      0.32      0.36        95
           7       0.27      0.36      0.31        36
           8       0.09      0.06      0.07        16

   micro avg       0.77      0.77      0.77      1497
   macro avg       0.40      0.41      0.40      1497
weighted avg       0.77      0.77      0.77      1497



## Support Vector Machine-[LINEAR FUNCTION]

In [122]:
# 1.Linear 
from sklearn.svm import SVC
svclassifier = SVC(kernel='linear')
svclassifier.fit(X, y)

SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
  decision_function_shape='ovr', degree=3, gamma='auto_deprecated',
  kernel='linear', max_iter=-1, probability=False, random_state=None,
  shrinking=True, tol=0.001, verbose=False)

In [123]:
y_pred = svclassifier.predict(X_test)

In [124]:
accuracy = svclassifier.score(X_test,y_test)
print(accuracy)

0.8864395457581831


#### THE ACCURACY OF THE DATA BY USING SUPPORT VECTOR MACHINE IN "LINEAR FUNCTION" IS 88.6%

In [125]:
from sklearn.metrics import classification_report, confusion_matrix
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))

[[960   0   0   0   0   0   0]
 [  0  91  37   3   4   0   0]
 [  0  34 115   7   4   1   0]
 [  0   6  20  57   9   2   0]
 [  0   2   5  11  70   6   1]
 [  0   0   4   3   8  21   0]
 [  0   0   0   2   1   0  13]]
              precision    recall  f1-score   support

           0       1.00      1.00      1.00       960
           1       0.68      0.67      0.68       135
           2       0.64      0.71      0.67       161
           5       0.69      0.61      0.64        94
           6       0.73      0.74      0.73        95
           7       0.70      0.58      0.64        36
           8       0.93      0.81      0.87        16

   micro avg       0.89      0.89      0.89      1497
   macro avg       0.77      0.73      0.75      1497
weighted avg       0.89      0.89      0.89      1497



## RANDOM FOREST

In [126]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=1)

In [127]:
#from sklearn.linear_model import LogisticRegression 
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score 

#model = LogisticRegression()
model = RandomForestClassifier(n_estimators=100)

model.fit(X, y)

RandomForestClassifier(bootstrap=True, class_weight=None, criterion='gini',
            max_depth=None, max_features='auto', max_leaf_nodes=None,
            min_impurity_decrease=0.0, min_impurity_split=None,
            min_samples_leaf=1, min_samples_split=2,
            min_weight_fraction_leaf=0.0, n_estimators=100, n_jobs=None,
            oob_score=False, random_state=None, verbose=0,
            warm_start=False)

In [130]:
y_pred = model.predict(X_test)

In [128]:
#predicted_classes = model.predict(X)

In [131]:
accuracy = accuracy_score(y_test,y_pred)
print(accuracy)

1.0


In [ ]:
#accuracy = accuracy_score(y.values.flatten(),predicted_classes)
#print(accuracy)

#### THE ACCURACY OF THE DATA BY USING RANDOM TREE IS 100%